### PR TITLE
feat(update-tabs): Populate tabs with templates

### PR DESCRIPTION
**Pull request by tristandubbeld**
In this PR we add the right templates (titles only for now) to the right tabs. There are a few things that will be changed in the future, I've created XTD-100 to keep track of work that still needs to be done. The actual template cards will be created in a different PR, hence why we're only showing titles.

Closes XTD-53
Closes XTD-56

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/6909/commits/cfd9c10c735d5aba8e789bb604d971178a41319d is at https://pr6909.build.csb.dev/s/new.

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/Essentials.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    track('Create Sandbox Tab Open', { tab: title });
+  }, [title]);
+
+  return (
+    <div>
+      <h2>{title}</h2>
+      <div>
+        {templates.length > 0 ? (
+          templates.map(template => (
+            <div key={template.id}>{template.sandbox.title}</div>
```
Comment (tristandubbeld) [inline packages/app/src/app/components/CreateNewSandbox/CreateSandbox/Essentials.tsx:21]: Showing only the title now as the cards are still a WIP

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/QuickStart/QuickStart.tsx [hunk 2]
```diff
@@ -32,18 +31,13 @@ export const Create: React.FC<CreateProps> = ({ collectionId }) => {
 
   return (
     <>
+      {/* ❗️ TODO: Move header to CreateSandbox component */}
```
Comment (tristandubbeld) [inline packages/app/src/app/components/CreateNewSandbox/CreateSandbox/QuickStart/QuickStart.tsx:34]: This will be done in a different PR

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx [hunk 3]
```diff
@@ -98,65 +141,46 @@ export const CreateSandbox: React.FC<CreateSandboxProps> = ({
...
+                  >
+                    {essential.title}
+                  </Tab>
+                ))
+              : null}
+
+            {essentialState.state === 'loading' ? (
+              <div>
+                <div>todo skeletons</div>
+                <SkeletonText css={{ width: 100 }} />
```
Comment (tristandubbeld) [inline packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx:174]: Work in progress

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx [hunk 4]
```diff
@@ -45,28 +51,65 @@ interface CreateSandboxProps {
...
+  const isUser = user?.username === activeTeamInfo?.name;
+
+  const [essentialState, setEssentialState] = useState<EssentialsState>({
+    state: 'loading',
+  });
+
   const tab = useTabState({
     orientation: 'vertical',
-    selectedId: initialTab || 'create',
+    selectedId: initialTab || isRepositoriesPage ? 'import' : 'create',
```
Comment (alexnm) [inline packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx:85]: is this checking if you are on `/dashboard/repositories`, as in the new page where we show the v2 repos?

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx [hunk 5]
```diff
@@ -45,28 +51,65 @@ interface CreateSandboxProps {
...
   });
 
   useEffect(() => {
-    /* ❗️ TODO: We can probably take this out of the useEffect and put it into the useTabState selectedId */
-    if (location.pathname.includes('/repositories')) {
-      tab.select('Import');
+    async function getEssentials() {
```
Comment (alexnm) [inline packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx:89]: nitpick: can you move this to a helper function perhaps? or a custom hook like `useEssentialTemplates`?

**Review by alexnm — APPROVED** [review]

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx [hunk 6]
```diff
@@ -45,28 +51,65 @@ interface CreateSandboxProps {
...
   initialTab,
   isModal,
 }) => {
-  const { hasLogIn } = useAppState();
+  const { hasLogIn, activeTeamInfo, user } = useAppState();
   const actions = useActions();
 
+  const isRepositoriesPage = location.pathname.includes('/repositories');
+  const isDashboardPage = location.pathname.includes('/dashboard');
+  const isUser = user?.username === activeTeamInfo?.name;
```
Comment (olarclara) [inline packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx:77]: Can you match if the active team id is equal to the user's personal team id?

**Review by olarclara — COMMENTED** [review]
I'm unable to scroll the templates sidebar, is it going to be fixed on the facelift PRs?

6upc76-3000 preview csb app_dashboard_home_workspace=411542e0-c5fb-4ee8-85c9-65c8cfd18e52

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx (hunk 4 above)
Comment (tristandubbeld) [reply packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx:85]: Nope, it check if the user is on the `codesandbox.io/dashboard/repositories`, not the v2 version. I guess we should rename this to "synced sandboxes" page and update the url as well? What is the latest update / name for that?

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx (hunk 6 above)
Comment (tristandubbeld) [reply packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx:77]: Sadly we can't. I checked the data I have and the only other relation between the `activeTeam` and the `user` are the `user.id` and the `activeTeamInfo.creatorId`. Cant use the `creatorId`, because multiple teams can have the same creator. The only relation that is left is the `user.username` and the `activeTeamInfo.name` so I used that one.

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx (hunk 5 above)
Comment (tristandubbeld) [reply packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx]: [Sure can do!](https://github.com/codesandbox/codesandbox-client/pull/6909/commits/f1fc82673c303d08f3d630d8cea857f800823b2d)

**Comment by tristandubbeld** [comment]
> I'm unable to scroll the templates sidebar, is it going to be fixed on the facelift PRs?

Yep, nice catch @olarclara. I should have put that into the PR description more prominently. We need to overhaul some of the structure later and fix the styles to make sure these tabs can be more dynamic. Didn't want to put it in this PR for size reasons 👍

**Review by olarclara — APPROVED** [review]
